src/main/java/io/stargate/sgv2/jsonapi/service/bridge/executor/QueryExecutor.java

```diff
@@ -118,35 +104,98 @@ private Uni<QueryOuterClass.ResultSet> queryBridge(QueryOuterClass.Query query)
-    return cqlKeyspaceDescribeUni
-        .onItemOrFailure()
-        .transformToUni(
-            (cqlKeyspaceDescribe, error) -> {
-              if (error != null
-                  && (error instanceof StatusRuntimeException sre
-                      && sre.getStatus().getCode() == Status.Code.NOT_FOUND)) {
-                return Uni.createFrom()
-                    .failure(
-                        new RuntimeException(
-                            new JsonApiException(
-                                ErrorCode.NAMESPACE_DOES_NOT_EXIST,
-                                "The provided namespace does not exist: " + namespace)));
-              }
-              Schema.CqlTable cqlTable = null;
-              return Uni.createFrom()
-                  .item(
-                      cqlKeyspaceDescribe.getTablesList().stream()
-                          .filter(table -> table.getName().equals(collectionName))
-                          .findFirst());
-            });
+  protected Uni<Optional<TableMetadata>> getSchema(String namespace, String collectionName) {
+    KeyspaceMetadata keyspaceMetadata;
+    try {
+      keyspaceMetadata =
+          cqlSessionCache
+              .getSession()
+              .getMetadata()
+              .getKeyspaces()
+              .get(CqlIdentifier.fromCql("\"" + namespace + "\""));
+    } catch (Exception e) {
+      return Uni.createFrom().failure(e);
+    }
+    // if namespace does not exist, throw error
+    if (keyspaceMetadata == null) {
+      return Uni.createFrom()
+          .failure(
+              new JsonApiException(
+                  ErrorCode.NAMESPACE_DOES_NOT_EXIST,
+                  "The provided namespace does not exist: " + namespace));
+    }
+    // else get the table
+    return Uni.createFrom().item(keyspaceMetadata.getTable("\"" + collectionName + "\""));
+  }
+
+  /**
+   * Gets the schema for the provided namespace and collection name
+   *
+   * @param namespace - namespace
+   * @param collectionName - collection name
+   * @return TableMetadata
+   */
+  protected Uni<TableMetadata> getCollectionSchema(String namespace, String collectionName) {
+    Optional<KeyspaceMetadata> keyspaceMetadata;
+    if ((keyspaceMetadata = cqlSessionCache.getSession().getMetadata().getKeyspace(namespace))
+        .isPresent()) {
+      Optional<TableMetadata> tableMetadata = keyspaceMetadata.get().getTable(collectionName);
+      if (tableMetadata.isPresent()) {
+        return Uni.createFrom().item(tableMetadata.get());
+      }
+    }
+    return Uni.createFrom().nullItem();
   }
 
   private static byte[] decodeBase64(String base64encoded) {
     return Base64.getDecoder().decode(base64encoded);
   }
+
+  /**
+   * Gets the consistency level for the provided QueryOuterClass.Consistency
+   *
+   * @param consistency - QueryOuterClass.Consistency
+   * @return ConsistencyLevel
+   */
+  private ConsistencyLevel getConsistencyLevel(QueryOuterClass.Consistency consistency) {
```

Comment:
Can we change the config to return driver ConsistencyLevel?

Reply:
Fixed in this commit a0ca6e003e1451734ad7566d9dfb12dd2face134